src/machine/machine_mimxrt1062_i2c.go

```diff
@@ -0,0 +1,725 @@
+// +build mimxrt1062
+
+package machine
+
+// I2C peripheral abstraction layer for the MIMXRT1062
+
+import (
+	"device/nxp"
+	"errors"
+)
+
+const (
+	TWI_FREQ_BUS     = 24000000        // LPI2C root clock is on 24 MHz OSC
+	TWI_FREQ_100KHZ  = 100000          // StandardMode (100 kHz)
+	TWI_FREQ_400KHZ  = 400000          // FastMode (400 kHz)
+	TWI_FREQ_1MHZ    = 1000000         // FastModePlus (1 MHz)
+	TWI_FREQ_5MHZ    = 5000000         // UltraFastMode (5 MHz)
+	TWI_FREQ_DEFAULT = TWI_FREQ_100KHZ // default to StandardMode (100 kHz)
+)
+
+var (
+	errI2CWriteTimeout       = errors.New("I2C timeout during write")
+	errI2CReadTimeout        = errors.New("I2C timeout during read")
+	errI2CBusReadyTimeout    = errors.New("I2C timeout on bus ready")
+	errI2CSignalStartTimeout = errors.New("I2C timeout on signal start")
+	errI2CSignalReadTimeout  = errors.New("I2C timeout on signal read")
+	errI2CSignalStopTimeout  = errors.New("I2C timeout on signal stop")
+	errI2CAckExpected        = errors.New("I2C error: expected ACK not NACK")
+	errI2CBusError           = errors.New("I2C bus error")
+	errI2CNotConfigured      = errors.New("I2C interface is not yet configured")
+)
+
+// I2CConfig is used to store config info for I2C.
+type I2CConfig struct {
+	Frequency uint32
+	SDA       Pin
+	SCL       Pin
+}
+
+func (c I2CConfig) getPins() (sda, scl Pin) {
+	if 0 == c.SDA && 0 == c.SCL {
+		// default pins if none specified
+		return I2C_SDA_PIN, I2C_SCL_PIN
+	}
+	return c.SDA, c.SCL
+}
+
+type I2C struct {
+	Bus *nxp.LPI2C_Type
+
+	// these hold the input selector ("daisy chain") values that select which pins
+	// are connected to the LPI2C device, and should be defined where the I2C
+	// instance is declared (e.g., in the board definition). see the godoc
+	// comments on type muxSelect for more details.
+	muxSDA, muxSCL muxSelect
+
+	// these are copied from I2CConfig, during (*I2C).Configure(I2CConfig), and
+	// should be considered read-only for internal reference (i.e., modifying them
+	// will have no desirable effect).
+	sda, scl  Pin
+	frequency uint32
+
+	// auxiliary state data used internally
+	configured bool
+}
+
+type I2CDirection bool
+
+const (
+	DirectionWrite I2CDirection = false
+	DirectionRead  I2CDirection = true
+)
```

Comment:
These symbols are exported but only seem to be used internally, so should not be exported.

Reply:
unexported type and constants:

```go
type i2cDirection bool

const (
	directionWrite i2cDirection = false
	directionRead  i2cDirection = true
)
```